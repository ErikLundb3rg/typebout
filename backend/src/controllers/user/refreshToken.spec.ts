/* eslint-disable @typescript-eslint/no-explicit-any */
import { logout } from './index'
import { refreshToken } from './index'

describe('RefreshToken Controller', () => {
  let req: any = {}
  let res: any = {}

  beforeEach(() => {
    req = {}
    res = {}
    res.cookie = jest.fn()
  })

  test('gives back refreshToken and accessToken', async () => {
    req = {
      user: {
        userId: 1232313131
      }
    }
    expect(refreshToken(req, res)).toMatchSnapshot()
    expect(res.cookie).toHaveBeenCalled()
  })
})
