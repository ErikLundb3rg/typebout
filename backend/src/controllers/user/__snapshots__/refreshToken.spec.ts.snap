// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefreshToken Controller gives back refreshToken and accessToken 1`] = `
{
  "data": {
    "accessToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiIxMjMyMzEzMTMxIiwiaWF0IjoxNjc0MTQzNTYzLCJleHAiOjE2Nzg0NjM1NjN9.lA3Pb__ue79VrkZ9pNhdx3bjtOyk4AN1pTeKSNlL658",
  },
  "message": "Successfully updated the access token",
  "ok": true,
  "status": 200,
}
`;
